upgrades/steps_28_test.go

```diff
@@ -197,17 +246,92 @@ func (s *mockSteps28Suite) expectAgentConfigValueIAAS() {
 	s.mockAgentConfig.EXPECT().Value(agent.ProviderType).Return("IAAS").AnyTimes()
 }
 
+func (s *mockSteps28Suite) expectAgentConfigMachineTag() {
+	s.mockAgentConfig.EXPECT().Tag().Return(names.NewMachineTag("0")).AnyTimes()
+}
+
+func (s *mockSteps28Suite) expectAgentConfigUnitTag() {
+	s.mockAgentConfig.EXPECT().Tag().Return(names.NewUnitTag("test/0"))
+}
+
 func (s *mockSteps28Suite) patchClient() {
 	s.PatchValue(upgrades.GetUpgradeStepsClient, func(_ base.APICaller) upgrades.UpgradeStepsClient {
 		return s.mockClient
 	})
 }
 
-func (s *mockSteps28Suite) expectWriteTwoUniterState() {
-	writeMap := map[names.Tag]string{
-		s.tagOne: s.opStateOneYaml,
-		s.tagTwo: s.opStateTwoYaml,
+func (s *mockSteps28Suite) expectWriteTwoUniterState(c *gc.C) {
+	args := []params.SetUnitStateArg{{
+		Tag:          s.tagOne.String(),
+		UniterState:  &s.opStateOneYaml,
+		StorageState: &s.opStorOneYaml,
+	}, {
+		Tag:         s.tagTwo.String(),
+		UniterState: &s.opStateTwoYaml,
+	},
 	}
 	cExp := s.mockClient.EXPECT()
-	cExp.WriteUniterState(writeMap).Return(nil)
+	cExp.WriteUniterState(unitStateMatcher{c, args}).Return(nil)
+}
+
+type unitStateMatcher struct {
+	c        *gc.C
+	expected []params.SetUnitStateArg
+}
+
+func (m unitStateMatcher) Matches(x interface{}) bool {
+	obtained, ok := x.([]params.SetUnitStateArg)
+	m.c.Assert(ok, jc.IsTrue)
+	if !ok {
+		return false
+	}
+
+	if len(obtained) != len(m.expected) {
+		m.c.Fatalf("obtained diff len than expected, obtained: %s, expected %s",
+			spew.Sdump(obtained), spew.Sdump(m.expected))
```

Comment:
I need to remove spew here

Reply:
Yeah, I think if you just replace '%s' with `%#+v` it should be fine.

Reply:
Also, spew not being present in Gopkg.toml is why the linter complains

Reply:
done